compiler/rustc_target/src/spec/mipsel_unknown_none.rs

```diff
@@ -0,0 +1,32 @@
+// Bare MIPS32r2, little endian, softfloat
+// can be used for MIPS M4K core (e.g. on PIC32MX devices)
+
+use crate::spec::{LinkerFlavor, LldFlavor, RelocModel};
+use crate::spec::{PanicStrategy, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        llvm_target: "mipsel-unknown-none".to_string(),
+        target_endian: "little".to_string(),
+        pointer_width: 32,
+        target_c_int_width: "32".to_string(),
+        data_layout: "e-m:m-p:32:32-i8:8:32-i16:16:32-i64:64-n32-S64".to_string(),
+        arch: "mips".to_string(),
+        target_os: "none".to_string(),
+        target_env: String::new(),
+        target_vendor: String::new(),
+        linker_flavor: LinkerFlavor::Lld(LldFlavor::Ld),
+
+        options: TargetOptions {
+            cpu: "mips32r2".to_string(),
```

Comment:
Is it common to default to mips32r2 (and not something older for example) in a generic MIPS target? What do other compilers do here?

Reply:
The mipsel linux target of rustc use MIPS32r2 (as opposed e.g. to mipsisa32r6-unknown-linux-gnu). Thus it is logical to use MIPS32r2 for mipsel-unknown-none. I do not expect that we need older ISA variants for commercially available targets.